lib/oli_web/live/workspaces/instructor/index_live.ex

```diff
@@ -4,22 +4,30 @@ defmodule OliWeb.Workspaces.Instructor.IndexLive do
   alias Oli.Delivery.Sections
   alias OliWeb.Backgrounds
   alias OliWeb.Common.{Params, SearchInput}
-  alias OliWeb.Components.Delivery.Utils
   alias OliWeb.Icons
+  alias Oli.Delivery.Sections.{Enrollment, EnrollmentContextRole, Section}
+  alias Oli.Repo
 
   import Ecto.Query, warn: false
   import OliWeb.Common.SourceImage
 
   @default_params %{text_search: "", sidebar_expanded: true}
 
+  @platform_instructor_roles_ids [
+    Lti_1p3.Tool.PlatformRoles.get_role(:institution_instructor).id
+  ]
+
+  @context_instructor_roles_ids [
+    Lti_1p3.Tool.ContextRoles.get_role(:context_instructor).id
+  ]
+
   @impl Phoenix.LiveView
   def mount(_params, _session, %{assigns: %{current_user: current_user}} = socket)
       when not is_nil(current_user) do
     sections =
-      Sections.list_user_open_and_free_sections(current_user)
-      |> add_user_role(current_user)
+      current_user.id
+      |> sections_where_user_is_instructor()
```

Comment:
Unfortunately, we don't have tests for the logic of pulling the section, except for one that I found in

https://github.com/Simon-Initiative/oli-torus/blob/e8ffbeaa5732bda895bde6414d1295e52751057a/test/oli/delivery/sections_test.exs#L2368

Calling this new `sections_where_user_is_instructor/1` in test mentioned I would expected to have the same result but I don't. See the following snippet:


```
      [s1, s3, s2] =
        OliWeb.Workspaces.Instructor.IndexLive.sections_where_user_is_instructor(user.id)
        |> Enum.filter(fn s -> s.user_role == "instructor" end)

      assert s1.title == "Elixir"
      assert s3.title == "LiveView"
      assert s2.title == "Phoenix"

      [s1, s3, s2] = Sections.list_user_open_and_free_sections(user)

      assert s1.title == "Elixir"
      assert s3.title == "LiveView"
      assert s2.title == "Phoenix"
```

Is that expected?

Reply:
Yep, because `list_user_open_and_free_sections/1` returns all the sections for a user (enrolled as student, instructor, etc).

Reply:
@manelli Notice in the code snippet I provided that I added an Enum equivalent to what `filter_by_role(:instructor)` is doing to replicate the current behavior:

https://github.com/Simon-Initiative/oli-torus/blob/9f2888cb308bbbcd619e6c00cc7bbaff87fa65e4/lib/oli_web/live/workspaces/instructor/index_live.ex#L18-L22

We should expect the same result for both your query and the combination provided in the snippet, right?

Reply:
The function that should be piped to `Enum.filter(fn s -> s.user_role == "instructor" end)`  is
```elixir
   Sections.list_user_open_and_free_sections(current_user) 
   |> add_user_role(current_user) 
```

not `sections_where_user_is_instructor`

Reply:
Thanks for pointing that. I'll do another loop then.